src/tracing/trace.ts

```diff
@@ -0,0 +1,59 @@
+import * as Sentry from '@sentry/react'
+import { Span, SpanStatusType } from '@sentry/tracing'
+
+interface TraceMetadata {
+  /** Arbitrary data stored on a trace. */
+  data?: Record<string, unknown>
+  /** Indexed (ie searchable) tags associated with a trace. */
+  tags?: Record<string, string | number | boolean>
+}
+
+// These methods are provided as an abstraction so that users will not interact with Sentry directly.
+// This avoids tightly coupling Sentry to our instrumentation outside of this file, in case we swap services.
+interface TraceCallbackOptions {
+  traceChild<T>(name: string, callback: TraceCallback<T>, metadata?: TraceMetadata): Promise<T>
+  setTraceData(key: string, value: unknown): void
+  setTraceTag(key: string, value: string | number | boolean): void
+  setTraceStatus(status: number | SpanStatusType): void
+  setTraceError(error: unknown): void
+}
+type TraceCallback<T> = (options: TraceCallbackOptions) => Promise<T>
+
+function traceTransaction(transaction?: Span) {
```

Comment:
can we add more documentation on this file since it's introducing important new methods that will likely be used in a lot of places? 

Including js docs on traceTransaction method, and each input param would be helpful for clarity.